include/arch/structs.h

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright (c) BayLibre SAS
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/*
+ * The purpose of this file is to provide essential/minimal architecture-
+ * specific structure definitions to be included in generic kernel
+ * structures.
+ *
+ * The following rules must be observed:
+ *  1. arch/structs.h shall not depend on kernel.h both directly and
+ *     indirectly (i.e. it shall not include any header files that include
+ *     kernel.h in their dependency chain).
+ *  2. kernel.h shall imply arch/structs.h via kernel_structs.h , such that
+ *     it shall not be necessary to include arch/structs.h explicitly when
+ *     kernel.h is included.
+ */
+
+#ifndef ZEPHYR_INCLUDE_ARCH_STRUCTS_H_
+#define ZEPHYR_INCLUDE_ARCH_STRUCTS_H_
+
+#if !defined(_ASMLANGUAGE)
+
+#if defined(CONFIG_ARM64)
+#include <arch/arm64/structs.h>
+#else
+
+/* Default definitions when no architecture specific definitions exist. */
+
+/* Per CPU architecture specifics (empty) */
+struct _cpu_arch {
+};
+
+#endif
+
+/* typedefs to be used with GEN_OFFSET_SYM(), etc. */
+typedef struct _cpu_arch _cpu_arch_t;
```

Comment:
Just so someone gets the pedantry into the public:

* Checkpatch hates typedefs (though Zephyr specifically doesn't have clear rules)
* MISRA hates symbols and types with leading underscores (though I think we're maybe starting to move away from strict adherence here)
* The offset generator really should be able to handle a struct type natively, that's what it's for.  Seems like we should fix that at some point.